container/kvm/cache_internal_test.go

```diff
@@ -0,0 +1,205 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+package kvm
+
+import (
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/juju/juju/environs/imagedownloads"
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+)
+
+// internalCacheSuite is gocheck boilerplate.
+type internalCacheSuite struct{}
+
+var _ = gc.Suite(internalCacheSuite{})
+
+const imageContents = "fake img file"
+
+func (internalCacheSuite) TestUpdater(c *gc.C) {
```

Comment:
This test looks like it's doing a lot \o/
I think it'd be nicer and easier to maintain if it was split into several tests each focused on discreet operations.